pkgs/applications/editors/emacs/default.nix

```diff
@@ -67,11 +61,17 @@ stdenv.mkDerivation rec {
              "--with-gif=no" "--with-tiff=no" ])
     ++ lib.optional withXwidgets "--with-xwidgets";
 
-  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.isDarwin && withX)
-    "-I${cairo.dev}/include/cairo";
-
-  preBuild = ''
+  preConfigure = lib.optionalString srcRepo ''
+    ./autogen.sh
+  '' + lib.optionalString srcLocal ''
     find . -name '*.elc' -delete
+  '' + ''
+    substituteInPlace lisp/international/mule-cmds.el \
+      --replace /usr/share/locale ${gettext}/share/locale
+
+    for i in Makefile.in {src,lib-src,leim}/Makefile.in; do
```

Comment:
Very good idea. That will make sure that all the Makefile.in that actually need it are covered. The following ones are currently not patched despite using /bin/pwd:
* ./Makefile.in
* ./nextstep/Makefile.in
* ./nt/Makefile.in

The following ones are patched but are not using /bin/pwd:
* leim/Makefile.in
* src/Makefile.in